sycl/test/xocc_tests/sdaccel_ports/vision/edge_detection/edge_detection.cpp

```diff
@@ -145,27 +116,23 @@ int main(int argc, char* argv[]) {
       // for HW. Currently pipelining the inner loop instead.
       for (size_t x = 1; x < width - 1; ++x) {
         for (size_t y = 1; y < height - 1; ++y) {
-            magX = 0; magY = 0;
-
-#ifdef XILINX
-            xilinx::pipeline([&] {
-#endif
-              for(size_t k = 0; k < 3; ++k) {
-                for(size_t l = 0; l < 3; ++l) {
-                  gI = k * 3 + l;
-                  pIndex =  (x + k - 1) + (y + l - 1) * width;
-                  magX += gX[gI] * pixel_rb[pIndex];
-                  magY += gY[gI] * pixel_rb[pIndex];
-                }
+          magX = 0; magY = 0;
+
+          xilinx::pipeline([&] {
+            for(size_t k = 0; k < 3; ++k) {
+              for(size_t l = 0; l < 3; ++l) {
+                gI = k * 3 + l;
+                pIndex =  (x + k - 1) + (y + l - 1) * width;
+                magX += gX[gI] * pixel_rb[pIndex];
+                magY += gY[gI] * pixel_rb[pIndex];
               }
-#ifdef XILINX
-            });
-#endif
+            }
+          });
 
-            // capping at 0xFF means no blurring of edges when it gets
-            // converted back to a char from an int
-            sum = std::abs(magX) + std::abs(magY);
-            pixel_wb[x + y * width] = (sum > 0xFF) ? 0xFF : (char)sum;
+          // capping at 0xFF means no blurring of edges when it gets
+          // converted back to a char from an int
+          sum = std::abs(magX) + std::abs(magY);
+          pixel_wb[x + y * width] = (sum > 0xFF) ? 0xFF : (char)sum;
```

Comment:
I am not sure this code is correct C++ in general, because  `char` might be signed and the you will have UB on the conversion.

After looking at the SYCL intrinsics, I prefer
```
pixel_wb[x + y * width] = min(sum, 0xFF);
```

Reply:
good idea.